runelite-api/src/main/java/net/runelite/api/events/EventBus.java

```diff
@@ -0,0 +1,139 @@
+import com.google.common.collect.ImmutableMultimap;
+import com.google.common.collect.Multimap;
+import com.google.common.eventbus.Subscribe;
+import java.lang.reflect.Method;
+import java.lang.reflect.Modifier;
+import lombok.Value;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class EventBus
+{
+	@Value
+	private static class Subscriber
+	{
+		private final Object object;
+		private final Method method;
+
+		void invoke(final Object arg) throws Exception
+		{
+			method.invoke(object, arg);
+		}
+	}
+
+	private ImmutableMultimap<Class, Subscriber> subscribers;
+
+	public synchronized void register(final Object object)
+	{
+		final ImmutableMultimap.Builder<Class, Subscriber> builder = ImmutableMultimap.builder();
+
+		if (subscribers != null)
+		{
+			builder.putAll(subscribers);
+		}
+
+		for (Class<?> clazz = object.getClass(); clazz != null; clazz = clazz.getSuperclass())
+		{
+			for (final Method method : clazz.getDeclaredMethods())
+			{
+				final Subscribe sub = method.getAnnotation(Subscribe.class);
+
+				if (sub == null)
+				{
+					continue;
+				}
+
+				assert method.getReturnType() == Void.TYPE : "@Subscribed methods cannot return a value";
+				assert method.getParameterCount() == 1 : "@Subscribed methods must take exactly 1 argument";
+				assert !Modifier.isStatic(method.getModifiers()) : "@Subscribed methods cannot be static";
+
+				final Class<?> parameterClazz = method.getParameterTypes()[0];
+
+				assert !parameterClazz.isPrimitive() : "@Subscribed methods cannot subscribe to primitives";
+
+				final String preferredName = "on" + parameterClazz.getSimpleName();
+				assert method.getName().equals(preferredName) : "Subscribed method " + method + " should be named " + preferredName;
+
+				method.setAccessible(true);
+
+				final Subscriber subscriber = new Subscriber(object, method);
+				builder.put(parameterClazz, subscriber);
+				log.debug("Registering {} - {}", parameterClazz, subscriber);
+			}
+		}
+
+		subscribers = builder.build();
+	}
+
+	public synchronized void unregister(final Object object)
+	{
+		if (subscribers == null)
+		{
+			return;
+		}
+
+		final Multimap<Class, Subscriber> map = HashMultimap.create();
+		map.putAll(subscribers);
+
+		for (Class<?> clazz = object.getClass(); clazz != null; clazz = clazz.getSuperclass())
+		{
+			for (final Method method : clazz.getDeclaredMethods())
+			{
+				final Subscribe sub = method.getAnnotation(Subscribe.class);
+
+				if (sub == null)
+				{
+					continue;
+				}
+
+				final Class<?> parameterClazz = method.getParameterTypes()[0];
+				map.remove(parameterClazz, new Subscriber(object, method));
+			}
+		}
+
+		subscribers = ImmutableMultimap.copyOf(map);
+	}
+
+	public void post(final Object event)
+	{
+		for (final Subscriber subscriber : subscribers.get(event.getClass()))
+		{
+			try
+			{
+				subscriber.invoke(event);
+			}
+			catch (Exception e)
+			{
+				log.warn("Uncaught exception in event subscriber", e);
```

Comment:
I kind of liked the exception handler thing guava has for the eventbus ctor, can you just mimic that?

Reply:
Do not seems very useful, but sure, updated